### PR TITLE
Format international phone numbers

### PR DESCRIPTION
**Pull request by ethax-ross**
We are changing the front-end to include the country dial-in code in international phone numbers.

If a candidate enters an international number we want to:

- Remove any non-digit character
- Prefix the number with `00`
- If the number has the UK dial-in code, swap it for a `0`

Telephone numbers for UK-based candidates should remain unchanged.

Due to the format change we can no longer easily use the telephone number itself to determine the destination; instead, we will assume if the candidate chooses a non-UK country then the telephone number is international.

### CODE REVIEW

File: GetIntoTeachingApi/Models/TeacherTrainingAdviserSignUp.cs [hunk 1]
```diff
@@ -164,6 +178,25 @@ private Candidate CreateCandidate()
...
+        {
+            var international = CountryId != LookupItem.UnitedKingdomCountryId;
+
+            if (international && telephone != null)
+            {
+                // Remove non-digit characters.
+                telephone = Regex.Replace(telephone, "[^0-9]", string.Empty);
+
+                // Prefix with 00.
+                telephone = $"00{telephone}";
```
Comment (peteryates) [inline GetIntoTeachingApi/Models/TeacherTrainingAdviserSignUp.cs:191]: Is it safe to assume that the person who typed it didn't already prepend the exit code?

Comment (ethax-ross) [reply GetIntoTeachingApi/Models/TeacherTrainingAdviserSignUp.cs:191]: Yeah we validate on the front-end to ensure the phone number starts with the country dial-in code

Comment (ethax-ross) [reply GetIntoTeachingApi/Models/TeacherTrainingAdviserSignUp.cs:191]: I didn't know that was called an exit code though - I'll update the comment 😄

**Review by peteryates — COMMENTED** [review]
Looks good, left one question

**Review by peteryates — APPROVED** [review]

File: GetIntoTeachingApi/Models/TeacherTrainingAdviserSignUp.cs [hunk 2]
```diff
@@ -35,7 +38,18 @@ public class TeacherTrainingAdviserSignUp
         public DateTime? DateOfBirth { get; set; }
         public string TeacherId { get; set; }
         public string DegreeSubject { get; set; }
-        public string AddressTelephone { get; set; }
+        public string AddressTelephone
+        {
```
Comment (jkempster34) [inline GetIntoTeachingApi/Models/TeacherTrainingAdviserSignUp.cs:42]: Is it worth using the arrow syntax to save some lines?

```c#
 get => return SanitizeTelephone(_addressTelephone);
 set => _addressTelephone = value;
```

Comment (ethax-ross) [reply GetIntoTeachingApi/Models/TeacherTrainingAdviserSignUp.cs:42]: Yeah that's a good shout, I'll maybe see if I can get the linter setup to catch that

**Pull request merged**